src/brand/bhyve/init

```diff
@@ -0,0 +1,320 @@
+# source. A copy of the CDDL is also available via the Internet at
+# http://www.illumos.org/license/CDDL.
+#
+# }}}
+
+# Copyright 2020 OmniOS Community Edition (OmniOSce) Association.
+
+import logging, os, subprocess, sys, time, re, getopt
+import xml.etree.ElementTree as etree
+from pprint import pprint, pformat
+
+zonexml = '/etc/zone.xml'
+uuidfile = '/etc/uuid'
+testmode = False
+
+try:
+    opts, args = getopt.getopt(sys.argv[1:], "tx:u:")
+except getopt.GetoptError:
+    print("init [-t] [-x <xml file>] [-u <uuid file>]")
+    sys.exit(2)
+for opt, arg in opts:
+    if opt == '-t':
+        testmode = True
+    elif opt == '-x':
+        zonexml = arg
+    elif opt == '-u':
+        uuidfile = arg
+
+if testmode:
+    logging.basicConfig(stream=sys.stdout, level=logging.DEBUG)
+else:
+    logging.basicConfig(filename='/tmp/init.log', filemode='a',
+        level=logging.DEBUG)
+
+# Default values
+opts = {
+    'acpi':         'on',
+    'vcpus':        '1',
+    'ram':          '1G',
+    'bootrom':      'BHYVE_RELEASE_CSM',
+    'bootorder':    'cd',
+    'hostbridge':   'i440fx',
+    'diskif':       'virtio-blk',
+    'netif':        'virtio-net-viona',
+    'type':         'generic',
+    'vnc':          'off',
+    'xhci':         '',
+    'console':      '/dev/zconsole',
+    'extra':        '',
+}
+
+aliases = {
+    'diskif': {
+        'virtio':       'virtio-blk',
+        'ahci':         'ahci-hd',
+    },
+    'netif': {
+        'virtio':       'virtio-net-viona',
+    },
+    'hostbridge': {
+        # Was wrongly used in some older scripts.
+        'intel':        'netapp',
+    },
+    'vnc': {
+        'on':   'unix=/tmp/vm.vnc',
+        'wait': 'wait,unix=/tmp/vm.vnc',
+    },
+}
+
+HOSTBRIDGE_SLOT = 0
+LPC_SLOT        = 1
+CDROM_SLOT      = 3
+BOOTDISK_SLOT   = 4
+DISK_SLOT       = 5
+NET_SLOT        = 6
+CDROM_SLOT2     = 7
+DISK_SLOT2      = 8
+VNC_SLOT        = 30
+LPC_SLOT_WIN    = 31
+
+try:
+    with open(uuidfile) as file:
+        uuid = file.read().strip()
+        logging.info('Zone UUID: {0}'.format(uuid))
+except:
+    uuid = None
+
+try:
+    cfg = etree.parse(zonexml)
+except:
+    logging.error('Could not parse {0}'.format(zonexml))
+    sys.exit(1)
+root = cfg.getroot()
+logging.info('Parsed {0}'.format(zonexml))
+
+##############################################################################
+
+def opt(tag):
+    global opts, root
+    try:
+        el = root.find('./attr[@name="{0}"]'.format(tag))
+        opts[tag] = el.get('value').strip()
+        logging.debug('Found custom {0} attribute - "{1}"'
+            .format(tag, opts[tag]))
+        if tag in aliases:
+            try:
+                opts[tag] = aliases[tag][opts[tag]]
+                logging.debug('  Alias expanded to {0}'.format(opts[tag]))
+            except KeyError:
+                pass
+    except:
+        pass
+
+def diskpath(arg):
+    if arg.startswith('/dev'):
```

Comment:
startswith('/')
Otherwise we can\t boot from iso.
```
ERROR:root:Error b'bhyve: Could not open backing file: /dev/zvol/rdsk//export/home/alp/distr/FreeBSD/FreeBSD-11.1-RELEASE-amd64-disc1.iso: No such file or directory\nCould not open backing file: No such file or directory\ndevice emulation initialization error: No such file or directory\n'
```

Reply:
The bootdisk attr does not need to be set to CDROM you set the bootorder to `cd` which means cdrom then drive or `dc`  which is for drive then cdrom. The bootdisk property is purely for Machines with multiple disks and should be optional otherwise. I have not tested without though.

```text
add fs
    set dir=/rpool/iso/debian-9.4.0-amd64-netinst.iso
    set special=/rpool/iso/debian-9.4.0-amd64-netinst.iso
    set type=lofs
    add options ro
    add options nodevices
end
add attr
    set name=cdrom
    set type=string
    set value=/rpool/iso/debian-9.4.0-amd64-netinst.iso
end
```

Reply:
I see this, but I'd prefer to be consistent in kvm and bhyve brand, so please, do this change.

Reply:
Oh, did not notice we had it differently. OmniOS has both looking for /dev not / in the diskpath. Looks like  both distros now use that attr for different things. Changed to our version.